src/haxelib/client/Main.hx

```diff
@@ -1283,36 +1289,40 @@ class Main {
 	function getDev( dir ) {
 		var path = File.getContent(dir + "/.dev").trim();
 		path = ~/%([A-Za-z0-9_]+)%/g.map(path,function(r) {
-			var env = Sys.getEnv(r.matched(1));
+			final env = Sys.getEnv(r.matched(1));
 			return env == null ? "" : env;
 		});
-		var filters = try Sys.getEnv("HAXELIB_DEV_FILTER").split(";") catch( e : Dynamic ) null;
+		final filters = try Sys.getEnv("HAXELIB_DEV_FILTER").split(";") catch( e : Dynamic ) null;
 		if( filters != null && !filters.exists(function(flt) return StringTools.startsWith(path.toLowerCase().split("\\").join("/"),flt.toLowerCase().split("\\").join("/"))) )
 			throw "This .dev is filtered";
 		return path;
 	}
 
 	function list() {
-		var rep = getRepository();
-		var folders = FileSystem.readDirectory(rep);
-		var filter = paramOpt();
-		if ( filter != null )
-			folders = folders.filter( function (f) return f.toLowerCase().indexOf(filter.toLowerCase()) > -1 );
-		var all = [];
+		final rep = getRepository();
+		final folders = {
```

Comment:
Just to mention a counterpoint, this one I actually like because it hides `filter` in a sub-scope, which is good because we don't need it afterwards. I'll have to get used to these nested `final something = { final something = }`, but I don't think it's a bad pattern per-se (I use it in OCaml too).